Missing secret fails silently

I'm guessing this refers to:

```
 Failed the docker starting procedure for bq-upload-user-sessions#BQLoad_DistCpFromOnPremise_UserSessionsAggregatedBySetType#2017-02-28
   java.lang.reflect.UndeclaredThrowableException: null
   at com.sun.proxy.$Proxy29.start(Unknown Source)
   at com.spotify.styx.state.handlers.DockerRunnerHandler.dockerRunnerStart(DockerRunnerHandler.java:136)
   at com.spotify.styx.state.handlers.DockerRunnerHandler.transitionInto(DockerRunnerHandler.java:68)
   at com.spotify.styx.state.FanOutputHandler.transitionInto(FanOutputHandler.java:44)
   at com.spotify.styx.state.FanOutputHandler.transitionInto(FanOutputHandler.java:44)
   at com.spotify.styx.state.QueuedStateManager.lambda$process$5(QueuedStateManager.java:279)
   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
   at java.lang.Thread.run(Thread.java:745)
   Caused by: java.lang.reflect.InvocationTargetException: null
   at sun.reflect.GeneratedMethodAccessor147.invoke(Unknown Source)
   at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   at java.lang.reflect.Method.invoke(Method.java:497)
   at com.spotify.styx.monitoring.MeteredProxy.invoke(MeteredProxy.java:64)
   at com.sun.proxy.$Proxy29.start(Unknown Source)
   at com.spotify.styx.state.handlers.DockerRunnerHandler.dockerRunnerStart(DockerRunnerHandler.java:136)
   at com.spotify.styx.state.handlers.DockerRunnerHandler.transitionInto(DockerRunnerHandler.java:68)
   at com.spotify.styx.state.FanOutputHandler.transitionInto(FanOutputHandler.java:44)
   at com.spotify.styx.state.FanOutputHandler.transitionInto(FanOutputHandler.java:44)
   at com.spotify.styx.state.QueuedStateManager.lambda$process$5(QueuedStateManager.java:279)
   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
   at java.lang.Thread.run(Thread.java:745)
   Caused by: com.spotify.styx.docker.InvalidExecutionException: Referenced secret 'bq-upload-user-sessions-secret' was not found
   at com.spotify.styx.docker.KubernetesDockerRunner.lambda$start$0(KubernetesDockerRunner.java:107)
   at java.util.Optional.ifPresent(Optional.java:159)
   at com.spotify.styx.docker.KubernetesDockerRunner.start(KubernetesDockerRunner.java:103)
   at com.spotify.styx.docker.RoutingDockerRunner.start(RoutingDockerRunner.java:51)
   at sun.reflect.GeneratedMethodAccessor147.invoke(Unknown Source)
   at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   at java.lang.reflect.Method.invoke(Method.java:497)
   at com.spotify.styx.monitoring.MeteredProxy.invoke(MeteredProxy.java:64)
   at com.sun.proxy.$Proxy29.start(Unknown Source)
   at com.spotify.styx.state.handlers.DockerRunnerHandler.dockerRunnerStart(DockerRunnerHandler.java:136)
   at com.spotify.styx.state.handlers.DockerRunnerHandler.transitionInto(DockerRunnerHandler.java:68)
   at com.spotify.styx.state.FanOutputHandler.transitionInto(FanOutputHandler.java:44)
   at com.spotify.styx.state.FanOutputHandler.transitionInto(FanOutputHandler.java:44)
   at com.spotify.styx.state.QueuedStateManager.lambda$process$5(QueuedStateManager.java:279)
   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
   at java.lang.Thread.run(Thread.java:745)
```

Which in turn causes

```
   java.lang.NullPointerException: Argument shouldn't be null: 'message' argument in static method invocation: 'runError' in class com.spotify.styx.model.Event
   at com.spotify.styx.model.Event.runError(Event.java:141)
   at com.spotify.styx.state.handlers.DockerRunnerHandler.transitionInto(DockerRunnerHandler.java:82)
   at com.spotify.styx.state.FanOutputHandler.transitionInto(FanOutputHandler.java:44)
   at com.spotify.styx.state.FanOutputHandler.transitionInto(FanOutputHandler.java:44)
   at com.spotify.styx.state.QueuedStateManager.lambda$process$5(QueuedStateManager.java:279)
   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
   at java.lang.Thread.run(Thread.java:745)
```

Seems the `InvalidExecutionException` that is thrown is now wrapped up in reflection exceptions because of `MeteredProxy`.

--- a/styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MeteredProxy.java
+++ b/styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MeteredProxy.java
@@ -24,6 +24,7 @@ import com.spotify.styx.docker.DockerRunner;
 import com.spotify.styx.storage.Storage;
 import com.spotify.styx.util.Time;
 import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
 import java.lang.reflect.Proxy;
 import java.time.Instant;
@@ -61,7 +62,12 @@ public class MeteredProxy implements InvocationHandler {
     final String operation = method.getName();
 
     final Instant t0 = time.get();
-    final Object ret = method.invoke(delegate, args);
+    final Object ret;
+    try {
+      ret = method.invoke(delegate, args);
+    } catch (InvocationTargetException e) {
+      throw e.getTargetException();
+    }
     final long durationMillis = t0.until(time.get(), ChronoUnit.MILLIS);
 
     if (delegate instanceof Storage) {

--- a/styx-scheduler-service/src/test/java/com/spotify/styx/monitoring/MeteredProxyTest.java
+++ b/styx-scheduler-service/src/test/java/com/spotify/styx/monitoring/MeteredProxyTest.java
@@ -20,6 +20,8 @@
 
 package com.spotify.styx.monitoring;
 
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doThrow;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.verify;
 
@@ -29,9 +31,14 @@ import com.spotify.styx.util.Time;
 import java.time.Instant;
 import java.util.Arrays;
 import java.util.List;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
 
 public class MeteredProxyTest {
+
+  @Rule
+  public ExpectedException expect = ExpectedException.none();
 
   Instant now = Instant.now();
   Instant later = now.plusMillis(123);
@@ -61,5 +68,18 @@ public class MeteredProxyTest {
 
     verify(mock).cleanup("barbaz");
     verify(stats).dockerOperation("cleanup", 123);
+  }
+
+  @Test
+  public void surfaceExceptions() throws Exception {
+    DockerRunner mock = mock(DockerRunner.class);
+    DockerRunner proxy = MeteredProxy.instrument(DockerRunner.class, mock, stats, time);
+
+    doThrow(new RuntimeException("with message")).when(mock).cleanup(any());
+
+    expect.expect(RuntimeException.class);
+    expect.expectMessage("with message");
+
+    proxy.cleanup("foo");
   }
 }

